omniscidb/QueryEngine/CostModel/ExtrapolationModels/LinearRegression.h

```diff
@@ -0,0 +1,42 @@
+/*
+    Copyright (c) 2023 Intel Corporation
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+        http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+
+#include <armadillo>
+#include "ExtrapolationModel.h"
+
+namespace costmodel {
+
+class LinearRegression : public ExtrapolationModel {
+ public:
+  LinearRegression(const std::vector<Detail::Measurement>& measurement)
+      : ExtrapolationModel(measurement) {
+    buildRegressionCoefficients();
+  }
+  LinearRegression(std::vector<Detail::Measurement>&& measurement)
+      : ExtrapolationModel(std::move(measurement)) {
+    buildRegressionCoefficients();
+  }
+
+  size_t getExtrapolatedData(size_t bytes) override;
+
+ protected:
+  void buildRegressionCoefficients();
+  arma::mat buildFeaturesMatrix();
+  arma::vec buildTargets();
+
+  arma::vec w;
```

Comment:
We usually use underscores for private or protected members. When in doubt, consult with https://google.github.io/styleguide/cppguide.html. This applies to the whole PR.